### DISCUSSION TITLE
Help with Blue

### DISCUSSION THREAD

**Discussion by konsumer** [bangcorrupt/hacktribe#268] [Q&A] [answered]
Hi,
Pardon me, if this is the wrong place to ask, but I have a Electribe2 (blue) that I bought specifically to install hacktribe on, and I can't figure out how to get it working with the new synths + original sounds.

I updated to hacktribe, and it seems to work (I can switch oscillators and they work great) but on standard firmware (2.02 from korg website) it has drums and lots more samples. I want both the new stuff and the old stuff.

If I understand correctly, I need to boot into original firmware, then "EXPORT ALL AUDIO", and manage it in Oe2sSLE, moving it all to 500+, but on device, I see no option to "EXPORT ALL AUDIO" in menu.

Is there another step I need to do?

I'd be happy to write some start-to-finish docs for the blue, if I figure out how to set it up.

**Comment by konsumer** [comment]
I also tried instructions [here](https://github.com/bangcorrupt/hacktribe/wiki/Dump-Synth-PCM-Samples) and get thsi error:

```
DEBUG:root:Initialise SysEx
Traceback (most recent call last):
  File "/Users/konsumer/Downloads/hacktribe/scripts/dump_extracted_pcm.py", line 8, in <module>
    e = E2Sysex()
        ^^^^^^^^^
  File "/Users/konsumer/Downloads/hacktribe/scripts/e2sysex.py", line 31, in __init__
    self.inport = mido.open_input(port)
                  ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/mido/backends/backend.py", line 91, in open_input
    return self.module.Input(name, **self._add_api(kwargs))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/mido/ports.py", line 161, in __init__
    BasePort.__init__(self, name, **kwargs)
  File "/opt/homebrew/lib/python3.11/site-packages/mido/ports.py", line 86, in __init__
    self._open(**kwargs)
  File "/opt/homebrew/lib/python3.11/site-packages/mido/backends/rtmidi.py", line 123, in _open
    rtapi = _get_api_id(api)
            ^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/mido/backends/rtmidi.py", line 34, in _get_api_id
    return rtmidi.API_UNSPECIFIED
           ^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'rtmidi' has no attribute 'API_UNSPECIFIED'
```

**Reply by derlunz** [reply]
I ran into this one too when trying to extract pcm synth samples. According to this [mido issue](https://github.com/mido/mido/issues/145) you seem to have the wrong version of `rtmidi` installed. You'll need the version with slightly different name called `python-rtmidi` that is available via pip.

Try to:
- remove *rtmidi* using `pip uninstall rtmidi`
- install the version that works with mido: `pip install python-rtmidi`

This got me around this particular problem.

**Comment by konsumer** [comment]
I sort of got it working without the dump-script, but all the samples are out of order and incorrect (like the saxes are drums.) Here is what I did:

- Load hacktribe firmware for blue (use the header patch option)
- checkout  `synth-pcm-dump` branch
- Get official sampler firmware, patch it
- Install on device
- boot with no SD, plug SD in after it's booted, "EXPORT ALL SAMPLES" (in menu now!)
- Checkout `main`, patch official sampler firmware
- Update (goes in `KORG/electribe sampler` not `KORG/electribe`)
- Now, you have `e2sSample.all` in `Samples/`, so offset start index to 500, save & copy that to `KORG/hacktribe/Samples` and boot

It definitely seems closer, but inspecting that `e2sSample.all`, it's definitely wrong. Is there any other way to extract this?

**Comment by bangcorrupt** [comment, marked as answer]
The PCM samples from the Synth are not fully supported yet, Hacktribe won't load or dump the multisampled instruments correctly.  At boot, the samples are decompressed and sent to the DSP RAM, the `synth-pcm-dump` branch will also store a copy of the samples in CPU RAM.  We can use `dump_extracted_pcm.py` to copy the samples as a single block via sysex, there isn't a script to cut them up yet.  I think the error you get with the script is due to not having `python-rtmidi` installed (note: `rtmidi-python` is a different package). 

- Install python dependencies and check the script runs.
- Install Hacktribe `synth-pcm-dump` branch and reboot.
- Run `dump_extracted_pcm.py`.
- This should give you the synth samples as a single block.
- You will need to manually cut this up and pick the samples you want.
- Install Hacktribe `main` branch.

Support for the synth samples and custom multisampled instrument is planned for the future.

**Comment by konsumer** [comment]
This totally makes sense. I ended up getting a dump of the samples as WAV files (with extra meta), which seems to be close enough to hear the basic sounds, even though it seems a bit off, still (multilayer, I'm sure.)

I think a related & cool feature would be to allow the new synths to be placed at another offset, so you can keep all the built-in samples and all the demo patterns work. Since I am new to Electribe2, it would be very cool if all the example patterns worked, so I could use the new stuff, but still see the factory demo patterns.
